core/src/main/java/org/elasticsearch/cluster/routing/IndexShardRoutingTable.java

```diff
@@ -312,6 +316,32 @@ public ShardIterator primaryFirstActiveInitializingShardsIt() {
         return new PlainShardIterator(shardId, ordered);
     }
 
+    public ShardIterator replicaActiveInitializingShardIt() {
+        if (!replicas.isEmpty() && !replicas.get(0).active() && !replicas.get(0).initializing()) {
```

Comment:
I wouldn't rely on the order the replicas exist in for this behavior, note it works in the primary case cause there can be only one :)